app/models/medium.rb

```diff
@@ -12,14 +12,11 @@ class MissingPutFilePath < StandardError; end
   before_destroy :queue_medium_removal, unless: :external_link
 
   ALLOWED_EXPORT_CONTENT_TYPES = %w(text/csv).freeze
+  BLOCKED_CONTENT_TYPES = %r{html|javascript|css|spx|opus|ttf|tar|gz|tgz|bz2|tbz2|zip}
```

Comment:
Is there a way to add `exe` files to this list as well? 

I think i prefer an explicit list of mime types here instead of a regex, e.g. %w[text/html... etc)
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

Reply:
The more I wrote explicit BLOCKED_CONTENT_TYPES/mime types that should be blocked, the more I convinced myself that I prefer an ALLOW_LIST (with explicitly listed out mime types). Here is my reasoning: 

1. explicitly listed blocked lists are harder to catch someone trying to fake content types (eg. your `test/html-json`)...though I question if smuggling here is an issue since I was under the assumption Azure Blob Storage "handles this". (By "handles this" - I mean for eg. if someone uploads a text/html under a faked mime type azure will return the type as the true mime type to panoptes for medium upload anyway.)
2. Allow lists are stricter, which I believe is something we want. (But maybe that's just my preference)
3.  It would keep it consistent with what UI validation is doing which uses an Allow list as well. *

Point 3 was probably the biggest selling point that pushed me to this decision.
I'm sorry to ask of you again @camallen . Do you mind reviewing newest changes? I appreciate your feedback.

Reply:
Not a problem for another review @yuenmichelle1 

2 & 3. yes they are - if you are happy with this then it's 👍🏼 from me. The main issue with an allow list is whenever a client wants a new mime type we need to code it up and deploy. Apart from that i think consistency between apps is a great idea.

Re point 1. The client submits the mime type to the api for what file it wants to upload. We then 
1. create a media resource in the db with the supplied mime type (it's it's allowed) 
2. we then request a blob storage URL that allows the client to put a file to that blob storage location on our blob storage system - note this URL has a time expiry so it's not allowed to write forever to that blob storage location. 
3. the client then uploads directly to the blob storage provider using the time limited put URL
4. the media resource can now provide a get URL for the blob storage location for reading in clients

So at no point does the blob storage system (azure / s3 etc) validate the mime / content type of the user supplied files. We trust the client provides the correct one and we set the content type on the blob storage system accordingly https://github.com/zooniverse/panoptes/blob/84de373bf03296d75d62e4f80a6857439f82fce4/app/services/media_storage/azure_adapter.rb#L59

Note we also use the correct file extension for the supplied content type when we create the blob storage path, https://github.com/zooniverse/panoptes/blob/84de373bf03296d75d62e4f80a6857439f82fce4/app/services/media_storage/azure_adapter.rb#L28

so we need to validate the content types are ok for us either from the block or allow list approach.

Reply:
ah i see. thanks for the context 👍